data/pigui/modules/station-view/04-shipMarket.lua

```diff
@@ -215,69 +229,99 @@ function FormatAndCompareShips:compare_and_draw_column(desc, str, a, b)
-	v = self[key]
+	local v = self[key]
 	if nil == v then
 		v = self.def[key]
 	end
 	return v
 end
 
 function FormatAndCompareShips:draw_tonnage_cell(desc, key)
-	self:compare_and_draw_column( desc, Format.MassTonnes(self:get_value(key)), self:get_value(key), self.b:get_value(key) )
+	self:compare_and_draw_column( desc, self:get_value(key), self.b:get_value(key), Format.MassTonnes )
 end
 
-function FormatAndCompareShips:draw_accel_cell(desc, thrustKey, massKey, multiplier)
+function FormatAndCompareShips:draw_accel_cell(desc, thrustKey, massKey, multiplier, inverse)
 	local accelA = multiplier * self.def.linearThrust[thrustKey] / (-9.81*1000*(self:get_value(massKey)))
 	local accelB = multiplier * self.b.def.linearThrust[thrustKey] / (-9.81*1000*(self.b:get_value(massKey)))
-	self:compare_and_draw_column( desc, Format.AccelG(accelA), accelA, accelB )
+	if inverse then
+		accelA = -accelA
+		accelB = -accelB
+	end
+	self:compare_and_draw_column( desc, accelA, accelB, Format.AccelG )
 end
 
 function FormatAndCompareShips:draw_deltav_cell(desc, massNumeratorKey, massDenominatorKey)
-	local deltavA = self.def.effectiveExhaustVelocity * math.log( self:get_value(massNumeratorKey), self.b:get_value(massDenominatorKey) )
-	local deltavB = self.b.def.effectiveExhaustVelocity * math.log( self.b:get_value(massNumeratorKey), self.b:get_value(massDenominatorKey) )	
-	self:compare_and_draw_column( desc, string.format("%d km/s", deltavA / 1000), deltavA, deltavB )
+	local deltavA = self.def.effectiveExhaustVelocity * math.log( self:get_value(massNumeratorKey) / self.b:get_value(massDenominatorKey) )
+	local deltavB = self.b.def.effectiveExhaustVelocity * math.log( self.b:get_value(massNumeratorKey) / self.b:get_value(massDenominatorKey) )	
+
+	local function fmt( v )
+		return string.format("%d km/s", v / 1000)
+	end
+
+	self:compare_and_draw_column( desc, deltavA, deltavB, fmt )
 end
 
 function FormatAndCompareShips:draw_unformated_cell(desc, key)
-	self:compare_and_draw_column( desc, self:get_value(key), self:get_value(key),  self.b:get_value(key) )
+	self:compare_and_draw_column( desc, self:get_value(key),  self.b:get_value(key) )
 end
 
 function FormatAndCompareShips:draw_equip_slot_cell(desc, key)
-	self:compare_and_draw_column( desc, self.def.equipSlotCapacity[key], self.def.equipSlotCapacity[key], self.b.def.equipSlotCapacity[key] )
+	self:compare_and_draw_column( desc, self.def.equipSlotCapacity[key], self.b.def.equipSlotCapacity[key] )
 end
 
 function FormatAndCompareShips:draw_yes_no_equip_slot_cell(desc, key)
-	binary = self.def.equipSlotCapacity[key]
-	yes_no = "unknown"
-	if binary == 1 then
-		yes_no = l.YES
-	elseif binary == 0 then
-		yes_no = l.NO
-	else
-		error("argument to yes_no not 0 or 1")
+
+	local function fmt( v )
+		local yes_no = "unknown"
+		if v == 1 then
+			yes_no = l.YES
+		elseif v == 0 then
+			yes_no = l.NO
+		else
+			error("argument to yes_no not 0 or 1")
+		end
+		return yes_no
```

Comment:
This should be expressed as `function (v) return v == 1 and l.YES or l.NO end`. The internal error bookkeeping overly complicates the purpose and is the wrong place to precondition the inputs; the `error()` means that `"unknown"` will never be returned as part of existing behavior and thus can be dropped.